src/encrypt.ts

```diff
@@ -0,0 +1,191 @@
+// external modules
+import { createCipheriv } from 'crypto-browserify';
+
+// local
+import { Cipher, CipherGCM } from 'crypto';
+import intoStream from 'into-stream';
+import { Readable } from 'stream';
+import toBuffer from 'typedarray-to-buffer';
+import { toWebReadableStream } from 'web-streams-node';
+import {
+  PenumbraEncryptedFile,
+  PenumbraEncryptionOptions,
+  PenumbraFileWithID,
+} from './types';
+
+// utils
+import { emitProgress, toBuff } from './utils';
+import emitEncryptionCompletion from './utils/emitEncryptionCompletion';
+
+/* tslint:disable completed-docs */
+
+// external modules
+
+/**
+ * Encrypts a readable stream
+ *
+ * @param rs - A readable stream of encrypted data
+ * @param cipher - The crypto module's cipher
+ * @param contentLength - The content length of the file, in bytes
+ * @param url - The URL to read the encrypted file from (only used for the event emitter)
+ * @returns A readable stream of decrypted data
+ */
+export function encryptStream(
+  jobID: number,
+  rs: ReadableStream | Readable,
+  cipher: CipherGCM,
+  contentLength: number,
+  key: Buffer,
+  iv: Buffer,
+): ReadableStream {
+  console.log('job id', jobID);
+  console.log('content length', contentLength);
+  const stream: ReadableStream =
+    rs instanceof ReadableStream ? rs : toWebReadableStream(rs);
+  let totalBytesRead = 0;
+
+  // TransformStreams are supported
+  if ('TransformStream' in self) {
+    return stream.pipeThrough(
+      // eslint-disable-next-line no-undef
+      new TransformStream({
+        transform: async (chunk, controller) => {
+          const bufferChunk = toBuffer(chunk);
+
+          // Encrypt chunk and send it out
+          const encryptedChunk = cipher.update(bufferChunk);
+          controller.enqueue(encryptedChunk);
+
+          // Emit a progress update
+          totalBytesRead += bufferChunk.length;
+          emitProgress(
+            'encrypt',
+            totalBytesRead,
+            contentLength,
+            '[encrypted file]',
+          );
+
+          if (totalBytesRead >= contentLength) {
+            // const authTag = cipher.getAuthTag();
```

Comment:
Consider adding a [flush](https://streams.spec.whatwg.org/#dom-transformer-flush) method to call

```js
cipher.final(outputEncoding);
const cipher = cipher.getAuthTag();  cipher.final needs to be called first
```